io-engine/src/core/logical_volume.rs

```diff
@@ -52,3 +54,90 @@ pub struct LvolSpaceUsage {
     /// clone.
     pub allocated_bytes_snapshot_from_clone: Option<u64>,
 }
+
+/// Arguments for replica creation.
+pub struct ReplicaArgs {
+    pub(crate) name: String,
+    pub(crate) size: u64,
+    pub(crate) uuid: String,
+    pub(crate) thin: bool,
+    pub(crate) entity_id: Option<String>,
+}
+
+/// This interface defines the high level operations which can be done on a
+/// pool. Pool-Specific details should be hidden away in the implementation as
+/// much as possible, though we can allow for extra pool specific options
+/// to be passed as parameters.
+#[async_trait::async_trait(?Send)]
+pub trait Pool {
+    type Replica: Replica + Share + std::fmt::Debug;
+    type Error: Into<tonic::Status> + std::fmt::Display;
+
+    /// List all replicas which exist on this pool.
+    async fn replicas(&self) -> Result<Vec<Self::Replica>, Self::Error>;
+    /// Create a replica on this pool with the given arguments.
+    async fn create_repl(
+        &self,
+        args: ReplicaArgs,
+    ) -> Result<Self::Replica, Self::Error>;
+    /// Destroy the pool itself along with all its replicas.
+    async fn destroy(self) -> Result<(), Self::Error>;
+    /// Exports the volume group by unloading all logical volumes.
+    /// The pool will no longer be listable until it is imported again.
+    async fn export(self) -> Result<(), Self::Error>;
+}
+
+/// This interface defines the high level operations which can be done on a
+/// `Pool` replica. Replica-Specific details should be hidden away in the
+/// implementation as much as possible, though we can allow for extra pool
+/// specific options to be passed as parameters.
+/// A `Replica` is also a `LogicalVolume` and also has `Share` traits.
+#[async_trait::async_trait(?Send)]
+pub trait Replica: Share + LogicalVolume {
```

Comment:
Let's call this also something like `ReplicaOps` or `ReplicaActions`

Reply:
Believe it or not I almost named it ReplicaOps previously :)
I left it as Replica simply because we also have `LogicalVolume` rather than `LogicalVolumeOps`.
So I think we if rename one we have to rename the other one as well?

Reply:
Although `LogicalVolume` is more properties, so maybe it's ok then

Reply:
I'd say let's rename now while you're at it, reason being it's already getting too confusing about names now with lvm in picture. `Lvol` implying `LvsLvol` vs `LogicalVolume` trait, `lvs` implying lvm logical volumes Vs `Lvs` backend, and so on !!
If it's too much to accommodate in this PR, then you may create a new ticket to address that as soon as it can be done.

Reply:
I've renamed and moved, PTAL